compiler/rustc_middle/src/ty/util.rs

```diff
@@ -1007,6 +1007,7 @@ impl<'tcx> TypeFolder<TyCtxt<'tcx>> for OpaqueTypeExpander<'tcx> {
 
 impl<'tcx> Ty<'tcx> {
     /// Returns the `Size` for primitive types (bool, uint, int, char, float).
+    /// If the type is not primitive, will panic with message "non primitive type".
```

Comment:
```suggestion
    /// Returns the `Size` for primitive types (bool, uint, int, char, float), or panic.
```

The message from the panic isn't useful as part of the docstring. If you hit a panic message, then grepping will get you to the actual panic place. You'd have the path to it to begin with. When reading the docs, the panic message wouldn't be useful, and looking at the method body to see the panic conditions is enough.

Reply:
My main concern was with the error message in `int_size_and_signed` - it mentions a discriminant, while the method is dealing with integers. In my opinion, a message mentioning a `non-intieger type` would be better,
but I did not want to change any code in my first commit.

This information was also added to this function for consistency’s sake.